development/contracts/Auction_ERC20.sol

```diff
@@ -65,79 +67,73 @@ contract Auction_ERC20 {
     }
 
     modifier afterForkBeforeCalculation() {
-        require(block.timestamp >= fork_ts, "must be after fork");
-        require(!is_calculation_done, "price calculation already done");
+        require(block.timestamp >= forkTimestamp, "must be after fork");
+        require(!isCalculationDone, "price calculation already done");
         _;
     }
 
     modifier afterForkAfterCalculation() {
-        require(is_calculation_done, "must be after price calculation");
+        require(isCalculationDone, "must be after price calculation");
         _;
     }
 
     // ForkManager should call this on deployment and credit this contract with the bonus amount
-    function init(uint256 _bonus, uint256 _fork_ts) 
+    // Todo: maybe this should be a constructor?
+    function init(uint256 _bonus, uint256 _forkTimestamp) 
     external {
         require(forkmanager == address(0), "Already initialized");
         forkmanager = msg.sender;
         bonus = _bonus;
-        fork_ts = _fork_ts;
+        forkTimestamp = _forkTimestamp;
     }
 
     // ForkManager should lock the tokens before calling this
-    function bid(address owner, uint8 _bid, uint256 _amount) 
-        onlyForkManager
+    function bid(address owner, uint8 _bid, uint88 _amount) 
```

Comment:
We need this because this contract doesn't call `transferFrom()`, ie we do

tx1: ForkManager.bid() -> calls Auction.bid()

The alternative would be
tx1: ForkManager.approve(auction)
tx2: Action.bid() calls ForkManager.transferFrom()

Reply:
Only read the forkmanager contract now. 

I think its good to separate concerns as much as possible. Hence I like your alternative.

But maybe I would even go one step further and separate the forkmanager and the ERC20 implementation. Let's talk about it in the call.